### DISCUSSION TITLE
OpenOffice: Wrong file path

### DISCUSSION THREAD

**Discussion by mihail-cornev** [totumonline/totum-ru-issues-and-discussions#524] [Как это сделать?]
Здравствуйте,
Спасибо за инструкцию по установке openoffice в этом посте #501 
Всё получилось и работает, но... Какое-то время назад я создал топик #289 с вопросом о единой таблице для хранения всех файлов в системе, и вроде бы всё получилось и работает, все ссылается в одну таблицу. Установил OpenOffice, и появилась проблемка: если заливаю файл в общую таблицу файлов - то все работает, открывается, а если залить из другой таблицы в неё же, через временную, то возникает ошибка wrong path. Направьте пожалуйста где может быть проблема.

Прилагаю видео по ссылке, т.к. не могу загрузить на github:
https://youtu.be/BXCclqtxI-0

Может это связано с тем, что после добавлению через временную файл получает название вида "fls/229_1_file!tmp!_3.xlsx" - но как это исправить я ума не приложу.

**Comment by ttm-alexeypolunin** [comment]
Вы что-то странно добавляете его. Покажите код, который переносит файлы из временной в протую.

**Reply by ttm-alexeypolunin** [reply]
Вообще что-то хитрое сделали. В цикле поле с файлами как организовано? Оно берет данные откуда?

**Reply by mihail-cornev** [reply]
Кнопка **Загрузить** по которой загружаю файл имеет код действия:

```
= : linkToFileUpload(title: "Загрузить файлы"; limit: 50; code: $code; type: "*"; refresh: true)`
code:totum
=: linkToDataTable(table: 'tmp_upload_files'; title: "Загрузить"; width: 800; height: "80vh"; params: $row; refresh: true)
row: rowCreate(field: "h_insert" = $#input; field: "ttm_table" = $#nti; field: "ttm_cycle" = $#nci; field: "h_file_from" = #h_file_from)
```



Во временной таблице из поля insert делаю listreplace коду действия:

```
= : listReplace(list: $#nfv; action: $val; value: "value")
      val: insert(table: $#ntn; field: 'file' = $filesdata)
        filesdata: listCreate(item: $rowfile)
	  		rowfile: rowCreate(field: "name" = $#value[name]; field: "filestring" = $content)
           		content: fileGetContent(file: $#value[file])
```

И потом по кнопке Сохранить эти файлы улетают в таблицу заказа и в общую таблицу:

```
a1= : insertList(table: 'files_'; field: 'file' = $list; field: 'file_ttm_table' = #ttm_table; field: 'file_ttm_cycle' = #ttm_cycle; field: 'file_uploaded_by' = $#nu; field: 'file_from' = #h_file_from;  inserts: "inserts"; log: true)
		list: selectList(table: $#ntn; field: 'file')

a2= : insertList(table: $sel; cycle: #ttm_cycle; field: 'file' = $#inserts; log: true)
		sel: select(table: 'tables'; field: 'name'; where: 'id' = #ttm_table)
```

**Reply by mihail-cornev** [reply]
В цикле два поля:

одно типа селект - которое ссылается на общую таблицу.
`= : selectRowListForSelect(table: 'files_'; field: 'file_name'; preview: 'file'; order: 'id' desc)`

а второе типа файл
`= : select(table: 'files_'; field: 'file'; where: 'id' = #file)`


![Screenshot_5](https://github.com/totumonline/totum-ru-issues-and-discussions/assets/113771444/d826a3e8-d7b2-4bbb-96a3-8d391f8a1065)

**Reply by mihail-cornev** [reply]
Файлы получилось добавить, а проблема с ошибкой осталась.
Если загружать напрямую из простой таблицы - то всё работает, а если из другой таблицы, то уже не работает - выдаёт ошибку.

**Reply by mihail-cornev** [reply]
https://youtu.be/Q47hPaLF07Y
Видимо неправильно берутся файлы в таблице циклов.
